### ISSUE TITLE
[Big Confusion] How to Set DP Query Parameters?

### ISSUE THREAD

**Issue by SamuelGong** [tensorflow/federated#838]
Hi developers,

I am running something basically like [DP-FedAvg for EMNIST](https://github.com/tensorflow/federated/blob/master/tensorflow_federated/python/research/differential_privacy/emnist/run_federated.py), however, I saw poor convergence using default parameters (I trained over 2000 rounds yet got accuracy unstable and as low as 50%.):

* `clip` = `0.05`
* `noise_multiplier` = `1.3`
* `adaptive_clip_learning_rate` = `0`
* `target_unclipped_quantile` = `0.5`
* `clipped_count_budget_allocation` = `0.1`
* `per_vector_clipping` = `False`

The [tutorial](https://www.tensorflow.org/federated/api_docs/python/tff/utils/build_dp_query) here does not help much since I do not know the ideal range of the parameters for me to set, not even empirical trials.

So I want to know that

* Is DP that bad, or exactly, in order to guarantee enough privacy, I can only achieve performance like that?
* I don't know how much DP noise is necessary. Is there any reference for me to set the parameters so that accuracy will be impact the least? How does the thing work in your case?

Thank you for the attention.

**Comment by nightldj** [comment]
You can try if adaptive clipping (https://arxiv.org/abs/1905.03871) helps by trying the following parameter, 

clip = 1
noise_multiplier = 1
adaptive_clip_learning_rate = 0.01
target_unclipped_quantile = 0.8

Also try a light-weight grid search for some of the parameters like
learning_rate=0.01,0.1,1
clip=0.5,1
noise_multiplier=0.1,1

etc.

**Comment by SamuelGong** [comment]
Thanks for the comment! The paper is a good reference and I also benefit from the suggested parameters.
